libcontainer/standard_init_linux.go

```diff
@@ -85,9 +86,16 @@ func (l *linuxStandardInit) Init() error {
 
 	// initialises the labeling system
 	selinux.GetEnabled()
-	if err := prepareRootfs(l.pipe, l.config); err != nil {
+	if err := prepareRootfs(l.pipe, l.config, l.mountFiles); err != nil {
 		return err
 	}
+	// We don't need mountFiles anymore: close them all
```

Comment:
It seems that if `prepareRootfs` returns an error, mount files are never closed.

What about something like
```go
err := prepareRootfs(l.pipe, l.config, l.mountFiles)
closeFiles(l.mountFiles)
if err != nil {
   return err
}
```

(whether to introduce and use `closeFiles()`, or do the `for` in place is up to you)

Reply:
In fact, I think it doesn't matter if the files are not closed upon error :-\ the original code was probably correct.

Reply:
Do you mean because if we fail the process will stop executing and therefore no need to close the fds? I think closing them is cleaner, so changed it to your suggestion :)